dd-sdk-android-core/src/test/kotlin/com/datadog/android/core/DatadogCoreTest.kt

```diff
@@ -235,6 +235,94 @@ internal class DatadogCoreTest {
+        val mockUserInfoProvider = mock<MutableUserInfoProvider>()
+        testedCore.coreFeature.userInfoProvider = mockUserInfoProvider
+        testedCore.setUserInfo(id, name, email, fakeMutableUserProperties)
+
+        // When
+        fakeMutableUserProperties.keys.forEach { key ->
+            fakeMutableUserProperties[key] = forge.anAlphaNumericalString()
+        }
+
+        // Then
+        argumentCaptor<UserInfo> {
+            verify(mockUserInfoProvider).setUserInfo(capture())
+            assertThat(lastValue.additionalProperties).isEqualTo(fakeExpectedUserProperties)
+        }
+    }
+
+    @Test
+    fun `M use immutable values W setUserInfo { adding properties }()`(
+        forge: Forge,
+        @StringForgery(type = StringForgeryType.HEXADECIMAL) id: String,
+        @StringForgery name: String,
+        @StringForgery(regex = "\\w+@\\w+") email: String,
+        @MapForgery(
+            key = AdvancedForgery(string = [StringForgery(StringForgeryType.ALPHA_NUMERICAL)]),
+            value = AdvancedForgery(string = [StringForgery(StringForgeryType.ALPHA_NUMERICAL)])
+        ) fakeUserProperties: Map<String, String>
+    ) {
+        // Given
+        val fakeMutableUserProperties = fakeUserProperties.toMutableMap()
+        val fakeExpectedUserProperties = fakeUserProperties.toMap()
+        val mockUserInfoProvider = mock<MutableUserInfoProvider>()
+        testedCore.coreFeature.userInfoProvider = mockUserInfoProvider
+        testedCore.setUserInfo(id, name, email, fakeMutableUserProperties)
+
+        // When
+        repeat(forge.anInt(0, 10)) {
+            fakeMutableUserProperties[forge.anAlphabeticalString()] = forge.anAlphabeticalString()
+        }
+
+        // Then
+        argumentCaptor<UserInfo> {
+            verify(mockUserInfoProvider).setUserInfo(capture())
+            assertThat(lastValue.additionalProperties).isEqualTo(fakeExpectedUserProperties)
+        }
+    }
+
+    @Test
+    fun `M use immutable values W setUserInfo { removing properties }()`(
+        forge: Forge,
+        @StringForgery(type = StringForgeryType.HEXADECIMAL) id: String,
+        @StringForgery name: String,
+        @StringForgery(regex = "\\w+@\\w+") email: String,
+        @MapForgery(
+            key = AdvancedForgery(string = [StringForgery(StringForgeryType.ALPHA_NUMERICAL)]),
+            value = AdvancedForgery(string = [StringForgery(StringForgeryType.ALPHA_NUMERICAL)])
+        ) fakeUserProperties: Map<String, String>
+    ) {
+        // Given
+        val fakeMutableUserProperties = fakeUserProperties.toMutableMap()
+        val fakeExpectedUserProperties = fakeUserProperties.toMap()
+        val mockUserInfoProvider = mock<MutableUserInfoProvider>()
+        testedCore.coreFeature.userInfoProvider = mockUserInfoProvider
+        testedCore.setUserInfo(id, name, email, fakeMutableUserProperties)
+
+        // When
+        fakeMutableUserProperties.values.drop(forge.anInt(0, fakeMutableUserProperties.size))
```

Comment:
> [!CAUTION]
> Calling the `values.drop()` doesn't modify the map, only the list of values. You need to call `m.remove(forge.anElementFrom(m.keys))`

> [!NOTE]
> Even if the drop worked, here we would only drop the 'first' n elements. Meaning that you would not remove stuff in the 'middle' of the map. It's not that relevant as Maps aren't ordered, but in some cases it could have some importance.